src/interface/src/app/standalone/planning-areas/planning-areas.datasource.ts

```diff
@@ -8,21 +8,44 @@ import { QueryParamsService } from './query-params.service';
 export class PlanningAreasDataSource extends DataSource<PreviewPlan> {
   private readonly limit = 13;
 
-  private _dataStream = new Subject<PreviewPlan[]>();
+  private _dataStream = new BehaviorSubject<PreviewPlan[]>([]);
   private _loading = new BehaviorSubject(false);
+  private _hasFilters$ = new BehaviorSubject(false);
+  private _initialLoad$ = new BehaviorSubject(true);
 
   public count = 0;
   public pages = 0;
 
+  public sortOptions: Sort = this.queryParamsService.getInitialSortParams();
+  public pageOptions = this.queryParamsService.getInitialPageParams();
+  public searchTerm = this.queryParamsService.getInitialFilterParam();
+
+  /**
+   * Emits `true` if loading the first time or applying filters (where number of results change)
+   * `false` when done loading.
+   */
+  public initialLoad$ = this._initialLoad$.asObservable();
+
+  /**
+   * Emits `true` if loading data. Includes sorting or changing pages.
+   * `false` when done loading.
+   */
   public loading$ = this._loading.asObservable();
-  public initialLoad$ = new BehaviorSubject(true);
-  public noEntries$ = this._dataStream.pipe(
-    map((results) => results.length === 0)
-  );
 
-  public sortOptions: Sort = this.queryParamsService.getInitialSortParams();
+  /**
+   * Emits `true` if there are no results after loading data.
+   */
+  public noEntries$ = combineLatest([
+    this.initialLoad$,
+    this._dataStream.asObservable(),
+  ]).pipe(
+    map(([initialLoad, results]) => !initialLoad && results.length === 0)
+  );
 
-  public pageOptions = this.queryParamsService.getInitialPageParams();
+  /**
+   * Emits `true` if applying filters or searching
+   */
+  public hasFilters$ = this._hasFilters$.asObservable();
```

Comment:
@lastminutediorama this one also would be something like hasFiltersOrSearch .... any ideas on a better name?